arch/x86/real_mode.S

```diff
@@ -139,6 +139,15 @@ END_FUNC(_prot_to_real)
 
 .align 16
 .Lfrom_long_mode:
+    /* Use protected mode data selector */
+    mov   $__KERN_DS32, %eax
+    mov   %eax, %ds
+
+    /* Disable paging to enter protected mode */
+    mov   %cr0, %eax
+    and   $~(X86_CR0_PG | X86_CR0_WP), %eax
+    mov   %eax, %cr0
+
     /* Disable LME in EFER */
```

Comment:
According to the SDM, disabling EFER.LME isn't needed, as the CPU does that itself when setting CR0.PG=0, see 9.8.5.4, Switching Out of IA-32e Mode Operation:

> 2. Deactivate IA-32e mode by clearing CR0.PG = 0. This causes the processor to set IA32_EFER.LMA = 0. The MOV CR0 instruction used to disable paging and subsequent instructions must be located in an identity-mapped page.

Reply:
True, and it seems to still work this way. However, they also say (why??)

> 4. Disable IA-32e mode by setting IA32_EFER.LME = 0.

Reply:
EFER.LMA and EFER.LME are two separate bits. both should be cleared. haha I have on idea why

Reply:
Oh, snap! Misread LME as LMA! Mea culpa!